### PR TITLE
pygraphviz functions updated

### PR DESCRIPTION
**Pull request by sachinkum0009**
the functions of graphviz are changed to the functions of pygraphviz

### CODE REVIEW

File: ros2controlcli/package.xml [hunk 1]
```diff
@@ -18,8 +18,8 @@
   <depend>controller_manager</depend>
   <depend>controller_manager_msgs</depend>
   <exec_depend>rosidl_runtime_py</exec_depend>
-  <exec_depend>python-graphviz-pip</exec_depend>
-  <exec_depend>graphviz</exec_depend>
+  <exec_depend>python-pygraphviz-pip</exec_depend>
+  <!-- <exec_depend>pygraphviz</exec_depend> -->
```
Comment (bmagyar) [inline ros2controlcli/package.xml:22]:
```suggestion
  <exec_depend>python-pygraphviz</exec_depend>
```

File: ros2controlcli/package.xml [hunk 2]
```diff
@@ -18,8 +18,7 @@
   <depend>controller_manager</depend>
   <depend>controller_manager_msgs</depend>
   <exec_depend>rosidl_runtime_py</exec_depend>
-  <exec_depend>python-graphviz-pip</exec_depend>
-  <exec_depend>graphviz</exec_depend>
+  <exec_depend>python-pygraphviz</exec_depend>
```
Comment (bmagyar) [inline ros2controlcli/package.xml:21]:
```suggestion
  <exec_depend>python3-pygraphviz</exec_depend>
```

File: ros2controlcli/ros2controlcli/verb/view_controller_chains.py [hunk 3]
```diff
@@ -21,7 +21,8 @@
 
 from ros2controlcli.api import add_controller_mgr_parsers
 
-import graphviz
+import pygraphviz as pgz
+# from PIL import Image
```
Comment (bmagyar) [inline ros2controlcli/ros2controlcli/verb/view_controller_chains.py:25]: No commented code please

File: ros2controlcli/ros2controlcli/verb/view_controller_chains.py [hunk 4]
```diff
@@ -107,19 +110,23 @@ def show_graph(input_chain_connections, output_chain_connections, command_connec
...
+                       '{}:{}'.format("command_interfaces", 'command_end_' + command_connection))
 
-    s.attr(ranksep='2')
-    s.attr(rankdir="LR")
+    s.graph_attr.update(ranksep='2')
+    s.layout(prog='dot')
+    s.draw('/tmp/controller_diagram.gv.pdf', format='pdf')
     if visualize:
-        s.view()
+        # s.draw('g.png')
```
Comment (bmagyar) [inline ros2controlcli/ros2controlcli/verb/view_controller_chains.py:126]: Commented code

File: ros2controlcli/ros2controlcli/verb/view_controller_chains.py [hunk 5]
```diff
@@ -107,19 +110,23 @@ def show_graph(input_chain_connections, output_chain_connections, command_connec
...
-            s.edge('{}:{}'.format(controller_name, "command_start_" + command_connection),
-                   '{}:{}'.format("command_interfaces", 'command_end_' + command_connection))
+            s.add_edge('{}:{}'.format(controller_name, "command_start_" + command_connection),
+                       '{}:{}'.format("command_interfaces", 'command_end_' + command_connection))
 
-    s.attr(ranksep='2')
-    s.attr(rankdir="LR")
+    s.graph_attr.update(ranksep='2')
+    s.layout(prog='dot')
+    s.draw('/tmp/controller_diagram.gv.pdf', format='pdf')
```
Comment (bmagyar) [inline ros2controlcli/ros2controlcli/verb/view_controller_chains.py:124]: Why draw it here and not below? That if looks very strange w only `pass` insode

**Review by bmagyar — CHANGES_REQUESTED** [review]
A few small comments

Also please for future PRs and for git in general try to avoid commit messages such as "minor changes", "fix review comments"

File: ros2controlcli/ros2controlcli/verb/view_controller_chains.py (hunk 3 above)
Comment (sachinkum0009) [reply ros2controlcli/ros2controlcli/verb/view_controller_chains.py:25]: will fix it

**Review by bmagyar — APPROVED** [review]
Thanks very much!

**Comment by bmagyar** [comment]
for completeness I've uploaded a sample here:

[g.pdf](https://github.com/ros-controls/ros2_control/files/9598741/g.pdf)

yes there are some issues to work out but this PR should get us past the release block

**Pull request merged**